.github/workflows/build-appimage.yml

```diff
@@ -0,0 +1,81 @@
+name: Build FLINT AppImage
+on:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, ubuntu-20.04, ubuntu-22.04]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Clone and run the AppImage script
+        run: |
+          git clone https://github.com/ankitaS11/AppImageDataForFlint
+          cd AppImageDataForFlint
+          chmod +x ./AppImage_FLINT_Build.sh
+          ./AppImage_FLINT_Build.sh
+          mv /home/runner/Downloads/FLINT/Source/build/FLINT-*.AppImage /home/runner/FLINT-${{ matrix.os }}.AppImage
+
+      - name: Cache artifacts
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-appimages
+        with:
+          path: /home/runner/FLINT-${{ matrix.os }}.AppImage
+          key: ${{ matrix.os }}-appimage
```

Comment:
How does caching play a role here?

Reply:
The cache is due to Github Actions' inability to share files across jobs. Every runner that completes a job, operates in its separate virtual environment, meaning we can't share Appimages between `build`, `test` and `release` jobs.

There are primarily two ways to solve this:

1. Use `upload-artifact` action to upload the appimages in the `build` job and the `download-artifact` action in `test` and `release` jobs. This action creates artifacts separately than the `release-action`.

2. Use the `cache` action. This allows us to cache the generated AppImages and retrieve them in other jobs using a single action.

In our case - we don't need the appimages available as an artifact since they're finally released in the `release` job anyway. We also aren't publishing any files generated as a side-effect during the build process. Also, the `cache` action is a bit faster than the upload/download-artifact actions according to my tests. So, I think `cache` action would be a better fit here.